drivers/regex/mlx5/mlx5_rxp.c

```diff
@@ -50,13 +102,150 @@ mlx5_regex_info_get(struct rte_regexdev *dev __rte_unused,
 	return 0;
 }
 
+/**
+ * Actual writing of RXP instructions to RXP via CSRs.
+ *
+ * @param ctx
+ *   The IBV context.
+ * @param rules
+ *   RXP instructions to be written.
+ * @param count
+ *   The number of instructions to be written.
+ * @param id
+ *   The selected engine.
+ *
+ * @return
+ *   0 on success, a negative errno value otherwise and rte_errno is set.
+ */
+static
+int rxp_write_rules_via_cp(struct ibv_context *ctx,
+                           struct mlx5_rxp_rof_entry *rules,
+                           int count, uint8_t id)
+{
+        int i, err = 0;
+        uint32_t tmp;
+
+        for (i = 0; i < count; i++) {
+                tmp = (uint32_t)rules[i].value;
+                err |= mlx5_regex_register_write(ctx, rxp_eng,
+                                        MLX5_RXP_RTRU_CSR_DATA_0, tmp);
+                tmp = (uint32_t)(rules[i].value >> 32);
+                err |= mlx5_regex_register_write(ctx, id,
+                                                 MLX5_RXP_RTRU_CSR_DATA_0 +
```

Comment:
please replace the mlx5_regex_register_xx with mlx5_devx_regex_register_write,
also please user replace err with ret or replace all ret with err